azurerm/internal/services/network/virtual_hub_connection_resource.go

```diff
@@ -170,36 +148,28 @@ func resourceArmVirtualHubConnectionRead(d *schema.ResourceData, meta interface{
 		return err
 	}
 
-	virtualHub, err := client.Get(ctx, id.ResourceGroup, id.VirtualHubName)
+	resp, err := client.Get(ctx, id.ResourceGroup, id.VirtualHubName, id.Name)
 	if err != nil {
-		if utils.ResponseWasNotFound(virtualHub.Response) {
-			log.Printf("[DEBUG] Virtual Hub %q was not found in Resource Group %q - so Connection %q can't exist - removing from state", id.Name, id.VirtualHubName, id.ResourceGroup)
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Virtual Hub Connection %q does not exist - removing from state", d.Id())
 			d.SetId("")
 			return nil
 		}
-
-		return fmt.Errorf("Error retrieving Virtual Hub %q (Resource Group %q): %+v", id.VirtualHubName, id.ResourceGroup, err)
+		return fmt.Errorf("reading Virtual Hub Connection %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
 	}
 
-	connection, err := findVirtualHubConnection(id.Name, virtualHub)
+	virtualHubResp, err := virtualHubClient.Get(ctx, id.ResourceGroup, id.VirtualHubName)
 	if err != nil {
-		return fmt.Errorf("Error retrieving Connection %q (Virtual Hub %q / Resource Group %q): %+v`", id.Name, id.VirtualHubName, id.ResourceGroup, err)
+		return fmt.Errorf("retrieving Virtual Hub %q (Resource Group %q): %+v", id.VirtualHubName, id.ResourceGroup, err)
 	}
-	if connection == nil {
-		log.Printf("[DEBUG] Connection %q was not found within Virtual Hub %q (Resource Group %q) - removing from state", id.Name, id.VirtualHubName, id.ResourceGroup)
-		d.SetId("")
-		return nil
-	}
-	if connection.HubVirtualNetworkConnectionProperties == nil {
-		return fmt.Errorf("Error retrieving Connection %q (Virtual Hub %q / Resource Group %q): `properties` was nil`", id.Name, id.VirtualHubName, id.ResourceGroup)
+	if virtualHubResp.ID == nil || *virtualHubResp.ID == "" {
+		return fmt.Errorf("Cannot read Virtual Hub %q (Resource Group %q) ID", id.VirtualHubName, id.ResourceGroup)
 	}
 
 	d.Set("name", id.Name)
-	d.Set("virtual_hub_id", virtualHub.ID)
+	d.Set("virtual_hub_id", virtualHubResp.ID)
 
-	if props := connection.HubVirtualNetworkConnectionProperties; props != nil {
-		d.Set("hub_to_vitual_network_traffic_allowed", props.AllowHubToRemoteVnetTransit)
-		d.Set("vitual_network_to_hub_gateways_traffic_allowed", props.AllowRemoteVnetToUseHubVnetGateways)
```

Comment:
we should probably still set these properties to `false` into the state for now

Reply:
this is still pending?

Reply:
These properties are always return `true`, as they are deprecated and has no effect whatever value they are set, we can totally ignore them in the diff via introducing a diff suppress function which always returns true?

Reply:
if someone removes these from their config this'll still be false though (and false for new resources) - so wouldn't that diff suppress func cause the deprecation not to be shown?

Reply:
Make sense.